### PR TITLE
Add option to pass cache bust param

### LINKED ISSUE #575: Add option to pass a cachebusting parameter
**Is it a duplicate question?**
no

**Is your feature request related to a problem? Please describe.**
Some websites use edge-caches such as **AWS cloudfront**.  
Meaning site HTML response is fully cached on some aws cloudfront pop server.  
Meaning "uptime kuma" will in essence be measuring uptime of random amazon cloudfront pop server in this case...
same if site uses varnish  caching proxy server for example.  
I would argue most folks would like to see origin uptimes instead. 

**Describe the solution you'd like**
Add an option to inject dynamically generated random query param so HTTP request breaks through cache and reaches origin. 
Add that dynamic query param cachebreaker to all urls:
`http://example.com?cb=RANDOMSTRING` 

**Describe alternatives you've considered**
im not aware of any.

**Additional context**
uptime kuma

### PR DESCRIPTION
**Pull request by dansullivan86**
- [x] I have read and understand the pull request rules.

# Description

Fixes #575 

## Type of change

Please delete any options that are not relevant.

- User interface (UI)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)
image

### CODE REVIEW

File: server/model/monitor.js
Comment (CommanderStorm) [inline server/model/monitor.js:465]: a shorter way would be the following:

```suggestion
                        const randomFloatString = Math.random().toString(36);
                        const cacheBust = randomFloatString.substring(2);
```

For equivalence (except in `cacheBurst` length) see 
image

File: src/lang/en.json
Comment (CommanderStorm) [inline src/lang/en.json:803]: What do you think about either of the following suggestions with adding `<code>uptime_kuma_cachebuster</code>` as a i18n-component template:

```suggestion
    "cacheBusterParam": "Add {0} as a parameter with random content ('cache buster')"
```

```suggestion
    "cacheBusterParam": "Add {0} as a cache buster parameter"
```

File: server/model/monitor.js
Comment (CommanderStorm) [inline server/model/monitor.js:466]: I have not checked if the code works for time reasons (I have an exam tomorrow, can check afterwards), but you have probably checked this anyway, so asking is "cheaper" overall:

Does this line mean that if a user has generic parameters (any parameter other than `uptime_kuma_cachebuster`) in his*her `url`, that they are deleted or is axios smarter?

**Review by CommanderStorm — CHANGES_REQUESTED** [review]

File: server/model/monitor.js
Comment (dansullivan86) [reply server/model/monitor.js:466]: Thankfully axios is smart here

File: src/lang/en.json
Comment (dansullivan86) [reply src/lang/en.json:803]: I think it makes a lot of sense to make it clear what he param is called here, are we happy with the text wrapping here?   
image

Comment (CommanderStorm) [reply src/lang/en.json:803]: thinking about it, I think we can even drop that it is a cache buster, given that that is literally written in the parameter name. Thoughts?

Comment (dansullivan86) [reply src/lang/en.json:803]: Thats true 
image

Comment (CommanderStorm) [reply src/lang/en.json:803]: Could you utilise the `i18n-t`-component to make this into `<code>uptime_kuma_cachebuster</code>`?

Should we add a helptext below it, or is cache busting a common enough thing that average users know what that means

File: server/model/monitor.js
Comment (CommanderStorm) [reply server/model/monitor.js]: Btw, you as the author can resolve discussions, making this clearer to other reviewers what was addressed and what is an open discussion ;)

File: src/lang/en.json
Comment (dansullivan86) [reply src/lang/en.json:803]: Apologies I missed your original meaning about the i18n component.  How about this: 
image

File: src/lang/en.json
Comment (CommanderStorm) [inline src/lang/en.json:804]:
```suggestion
    "cacheBusterParam": "Add the {0} parameter",
    "cacheBusterParamDescription": "Randomly generated parameter to skip caches."
```

Comment (dansullivan86) [reply src/lang/en.json]: Works for me

File: server/model/monitor.js
Comment (Computroniks) [inline server/model/monitor.js:282]: Is a new method really needed here? Given it is only used once I think, to me it seems cleaner just to put `Boolean(this.cacheBust)` where ever it is called.

```suggestion
```

File: server/model/monitor.js
Comment (Computroniks) [inline server/model/monitor.js:146]:
```suggestion
            cacheBust: Boolean(this.cacheBust),
```

File: server/model/monitor.js
Comment (dansullivan86) [reply server/model/monitor.js:282]: I did it this way to be consistent with all the other Boolean values in this file. 
`/**
     * Is the TLS expiry notification enabled?
     * @returns {boolean}
     */
    isEnabledExpiryNotification() {
        return Boolean(this.expiryNotification);
    }

    /**
     * Parse to boolean
     * @returns {boolean}
     */
    getIgnoreTls() {
        return Boolean(this.ignoreTls);
    }

    /**
     * Parse to boolean
     * @returns {boolean}
     */
    isUpsideDown() {
        return Boolean(this.upsideDown);
    }
`

File: server/database.js
Comment (louislam) [inline server/database.js:82]: Need to use knex migrations

https://github.com/louislam/uptime-kuma/tree/master/db/knex_migrations

File: server/model/monitor.js
Comment (CommanderStorm) [inline server/model/monitor.js:299]:
```suggestion
     * @returns {boolean} if cachebusting is enabled
```

File: server/server.js
Comment (CommanderStorm) [inline server/server.js:1446]:
```suggestion
```

**Review by CommanderStorm — COMMENTED** [review]
This comment fixes two linting issues.
Issue is still blocked on the missing db-migration issue noted above.

File: db/knex_migrations/2024-08-24-000-add-cache-bust.js
Comment (CommanderStorm) [inline db/knex_migrations/2024-08-24-000-add-cache-bust.js:6]: fixes linting error

```suggestion
      .alterTable("monitor", function (table) {
          table.boolean("cache_bust").notNullable().defaultTo(false);
      });
```

**Review by CommanderStorm — APPROVED** [review]
Thanks for the feature! 🎉 



> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>

File: db/knex_migrations/2024-08-24-000-add-cache-bust.js
Comment (CommanderStorm) [inline db/knex_migrations/2024-08-24-000-add-cache-bust.js:5]: fixed the linting error properly

```suggestion
        .alterTable("monitor", function (table) {
            table.boolean("cache_bust").notNullable().defaultTo(false);
        });
```

**Comment by Root-FTW** [comment]
I don't see the option, how can I activate it?

image

**Comment by CommanderStorm** [comment]
What version are you running? (Are you running the beta)
This is a V2 only feature.

**Pull request merged**
